src/k2/dto/K23SI.h

```diff
@@ -210,6 +212,41 @@ struct K23SIWriteResponse {
     K2_PAYLOAD_EMPTY;
 };
 
+enum class K23SIQueryOp : uint8_t {
+    EQ,
+    NEQ,
+    GT,
+    GTE,
+    LT,
+    LTE,
+    STARTS_WITH
+};
+
+struct K23SIQueryRequest {
```

Comment:
it would be nice if we could add some convenience API to build predicates, something like qreq.addFilter("name", K23SIQueryOp.STARTS_WITH, "Jo");

... I just saw you added that below. ANy reason not to have the API on this struct?